guides/common/modules/con_understanding-puppet-architecture.adoc

```diff
@@ -0,0 +1,19 @@
+[id="Understanding_puppet_architecture_{context}"]
+= Understanding Puppet Architecture
+
+Puppet uses client-server architecture with managed hosts running the Puppet agent and {Project} or any {SmartProxies} acting as a Puppet server.
+
+The Puppet agent collects _facts_ about a host and reports them to the Puppet server on each run.
+The Puppet server compares the reported _facts_ with the system state defined for the host.
+If the Puppet server finds differences (_drifts_) between the defined state and the actual state,
+it instructs the Puppet agent to enforce correction of the host's state. The Puppet agent then reports correction results back to {Project}.
+
+The system state of a host is defined by means of a catalog. A catalog is compiled from Puppet manifests of one or more Puppet modules assigned to the host.
+A Puppet module is a collection of classes, manifests, resources, files, and templates. The Puppet modules work as components of host configuration definitions.
+
+{Project} supports easy re-distribution of Puppet modules, which can be parametrized using Smart Class parameters if the module supports the use of parameters.
```

Comment:
`easy re-distribution` that holds true only for Katello < 4, the puppet modules content type was dropped with Pulp 3 upgrade